workers/unity/Packages/com.improbable.gdk.tools/DevAuthTokenUtils.cs

```diff
@@ -33,18 +43,48 @@ private static void Generate()
 
             try
             {
-                if (Json.Deserialize(receivedMessage).TryGetValue(JsonDataKey, out var jsonData) &&
+                var deserializedMessage = Json.Deserialize(receivedMessage);
+                if (deserializedMessage.TryGetValue(JsonDataKey, out var jsonData) &&
                     ((Dictionary<string, object>) jsonData).TryGetValue(TokenSecretKey, out var tokenSecret))
                 {
                     devAuthToken = (string) tokenSecret;
                 }
+                else
+                {
+                    throw new Exception(
+                        $@"{(deserializedMessage.TryGetValue(ErrorKey, out var errorMessage)
+                            ? errorMessage
+                            : string.Empty)}");
+                }
             }
             catch (Exception e)
             {
                 Debug.LogError($"Unable to generate Dev Auth Token. {e.Message}");
-                return;
+                return false;
             }
 
+            Debug.Log($"Saving token {devAuthToken} to Editor Preferences.");
```

Comment:
on second thought: let's not print out the token secret in case people share editor logs with each other. Just say "Saving token to ..."
also, you are saving it to the Player Preferences, not the Editor preferences